holoviews/core/dimension.py

```diff
@@ -1406,18 +1406,25 @@ def options(self, *args, **kwargs):
                              "or explicitly define the type, e.g. "
                              "obj.options({'Image': {'cmap': 'viridis'}}). "
                              "Supplying both formats is not supported.")
+        elif args and isinstance(args[0], dict):
+            options = [Options(spec, **kws) for spec,kws in args[0].items()]
```

Comment:
If there is more than one arg this should either warn or process all the args, silently dropping other args doesn't seem great.

Reply:
Agreed though the use of `args[0]` wasn't introduced in this PR. I'll add the warnings as suggested.

Reply:
Unfortunately, it isn't simple to fix due to the way `opts` and `options` invoke each other. Valid signatures would issue such a warning.

Reply:
Not sure I follow, when would passing in multiple dictionaries as kwargs ever occur?

Reply:
I can probably warn for this one condition when dicts are involved but not for other cases where `args[0]` is used and potentially dropping other positional arguments.

Reply:
Addressed in 9b9240bc1